db/db_impl/db_impl.cc

```diff
@@ -166,7 +166,7 @@ DBImpl::DBImpl(const DBOptions& options, const std::string& dbname,
       mutable_db_options_(initial_db_options_),
       stats_(immutable_db_options_.stats),
       mutex_(stats_, immutable_db_options_.clock, DB_MUTEX_WAIT_MICROS,
-             immutable_db_options_.use_adaptive_mutex),
+             immutable_db_options_.use_adaptive_mutex, &bg_cv_),
```

Comment:
What about here: 

```
#ifdef COERCE_CONTEXT_SWITCH
      mutex_(stats_, immutable_db_options_.clock, DB_MUTEX_WAIT_MICROS,
             immutable_db_options_.use_adaptive_mutex, &bg_cv_),
#else  // COERCE_CONTEXT_SWITCH
      mutex_(stats_, immutable_db_options_.clock, DB_MUTEX_WAIT_MICROS,
             immutable_db_options_.use_adaptive_mutex),
#endif  // COERCE_CONTEXT_SWITCH
```

And then have a new InstrumentedMutex constructor that takes `InstrumentedCondVar* bg_cv` conditionally compiled when `#ifdef COERCE_CONTEXT_SWITCH`?

Reply:
Sounds good!

Reply:
Fixed